src/instance.ts

```diff
@@ -358,16 +358,19 @@ function setupAfterCompile(compiler, instanceName, forkChecker = false) {
             }
 
             let diagnostics = state.ts.getPreEmitDiagnostics(state.program);
-            let emitError = (err) => {
+            let emitError = (msg) => {
                 if (compilation.bail) {
-                    console.error('Error in bail mode:', err);
+                    console.error('Error in bail mode:', msg);
                     process.exit(1);
                 }
-                compilation.errors.push(new Error(err));
+                compilation.errors.push(new Error(msg));
             };
 
-            let errors = formatErrors(instanceName, diagnostics);
-            errors.forEach(emitError);
+            let {options: {ignoreDiagnostics}} = instance;
+            diagnostics
+                .filter(err => !ignoreDiagnostics || ignoreDiagnostics.indexOf(err.code) == -1)
+                .map(err => `[${ instanceName }] ` + formatError(err))
```

Comment:
`instanceName` was moved here. It kept me from having to make `formatError` effectively curried. It also seemed like a preferable separation of concerns since `formatError` has more to do with the diagnostic than the inner working of atl.